src/Symfony/Component/Yaml/Parser.php

```diff
@@ -750,54 +750,54 @@ private function parseValue(string $value, int $flags, string $context)
                 return Inline::parse($this->lexInlineSequence($cursor), $flags, $this->refs);
             }
 
-            $quotation = '' !== $value && ('"' === $value[0] || "'" === $value[0]) ? $value[0] : null;
-
-            // do not take following lines into account when the current line is a quoted single line value
-            if (null !== $quotation && self::preg_match('/^'.$quotation.'.*'.$quotation.'(\s*#.*)?$/', $value)) {
-                return Inline::parse($value, $flags, $this->refs);
-            }
+            switch ($value[0] ?? null) {
```

Comment:
IIRC, PHP 7.+ has optimizations for switch when the cases are all integers or all strings and the type matches. So using `?? ''` instead of `?? null` might be better to avoid a type mismatch in the switch.